### PR TITLE
 Add CBE form ucr

### PR DESCRIPTION
**Pull request by Rohit25negi**
This is the Prep work for Service Delivery dashboard. This code adds the cbe ucr to collect the data of CBE forms.

functional spec: https://docs.google.com/document/d/1g1ldGNoYPoNbFEdl2zpUdDncfrbJoJcOl7TpjcdEuMc/edit

tech Spec: https://docs.google.com/spreadsheets/d/1m6QplhblN7HbOULgUiKqv78At-W5d9RF3u7DYuHosn8/edit?ts=5c471779#gid=0&range=B7

@calellowitz @emord

### CODE REVIEW

File: custom/icds_reports/ucr/data_sources/cbe_form.json [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                      "type": "location_parent_id"
+                    }
+                  },
+                  "type": "location_parent_id"
+                }
+              },
+              "type": "location_parent_id"
+            }
+          },
+          "type": "location_parent_id"
```
Comment (emord) [inline custom/icds_reports/ucr/data_sources/cbe_form.json]: This is more readable if you use `ancestor_location`

**Review by emord — APPROVED** [review]

File: custom/icds_reports/ucr/data_sources/cbe_form.json [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            "name": "user_location_id"
+          },
+          "location_type": "state",
+          "location_property": "_id"
+        },
+        "is_primary_key": false,
+        "transform": {},
+        "is_nullable": true,
+        "type": "expression",
+        "column_id": "state_id"
```
Comment (Rohit25negi) [inline custom/icds_reports/ucr/data_sources/cbe_form.json]: @emord is it fine now? Used ancestor location to get the state form awc_id

**Comment by Rohit25negi** [comment]
@calellowitz  Can you also look into this one. As there is not harm in having an extra eye :P

File: custom/icds_reports/ucr/data_sources/cbe_form.json [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+          "expression": {
+            "datatype": "datetime",
+            "type": "property_path",
+            "property_path": [
+              "form",
+              "meta",
+              "timeEnd"
+            ]
+          },
+          "type": "root_doc"
```
Comment (calellowitz) [inline custom/icds_reports/ucr/data_sources/cbe_form.json]: does this need to be `root_doc`?

File: custom/icds_reports/ucr/data_sources/cbe_form.json [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        "is_nullable": true,
+        "column_id": "submitted_on",
+        "type": "expression"
+      }
+    ],
+    "named_expressions": {
+      "user_location_id": {
+        "type": "icds_user_location",
+        "user_id_expression": {
+          "type": "root_doc",
```
Comment (calellowitz) [inline custom/icds_reports/ucr/data_sources/cbe_form.json]: same `root_doc` question

**Review by calellowitz — COMMENTED** [review]
I think this looks, two small questions but they arent blocking. I think those UCRs will probably work but not sure if those two extra parts of the expression are necessary.

File: custom/icds_reports/ucr/data_sources/cbe_form.json (hunk 3 above)
Comment (Rohit25negi) [reply custom/icds_reports/ucr/data_sources/cbe_form.json]: yeah U are right I dont need that and resolved.

Unrelated to this Pr. root_doc is used to find a tag from the root of the xml doc right? and what is the root_doc then if not form? is it<xml>?

**Pull request merged**